### ISSUE TITLE
tour: Bug in Internet Explorer 11

### ISSUE THREAD

**Issue by ndrewnee** [golang/go#12084]
Context: http://tour.golang.org/
Browser: Internet Explorer 11
Steps:
1. Go to http://tour.golang.org/
2. Click to "Next"
Expected Result: Go to next lesson
Result: Goes to next section, not lesson
P.S And button "Previous" doesn't work at all

**Comment by ianlancetaylor** [comment]
CC @adg @campoy

**Comment by andresteingress** [comment]
Happens in IE 10 too. When clicking the > arrow for the first time, `$routeParams.pageNumber` at some point contains a `String` instead of the page number from the URI, so the page number becomes `NaN` and thing start to fall apart.

**Comment by campoy** [comment]
I've investigated this but it is pretty hard to follow.

For some reason the url change event is fired multiple times, one of those
while the url is empty and that's why it fails to parse the page number.

If anyone has a good understanding of the specificities of angular on IE I
could definitely use some help debugging.

On Fri, Aug 14, 2015 at 11:49 AM, Andre Steingress <notifications@github.com

**Comment by minux** [comment]
This CL has been reverted, re-opening.
